index.js

```diff
@@ -153,39 +210,68 @@ function updateDescription(newDescription) {
   description.textContent = newDescription;
 }
 
-function changeInterfaceColors(buttonColor, shadowColor, ellipseColor, hoverColor) {
-  ellipse.style.setProperty('--bg-color', ellipseColor);
-  prevBtn.style.setProperty('--bg-color', buttonColor);
-  prevBtn.style.setProperty('--shadow-bg', shadowColor);
-  prevBtn.style.setProperty('--hover-color', hoverColor);
-  nextBtn.style.setProperty('--bg-color', buttonColor);
-  nextBtn.style.setProperty('--shadow-bg', shadowColor);
-  nextBtn.style.setProperty('--hover-color', hoverColor);
-  button.style.setProperty('--bg-orange', buttonColor);
-  button.style.setProperty('--shadow-bg', shadowColor);
-  button.style.setProperty('--hover-color', hoverColor);
-
-  items.forEach(item => {
-    item.style.setProperty('--hover-color', hoverColor);
-  });
+class Dishes extends HTMLElement {
+  constructor() {
+    super();
+    const image = this.getAttribute('image');
+    const html = `
+      <div class="carousel-food-spin__main js-carousel-food-spin-dish">
+        <a href="#" aria-label="click to see the selected dish">
+          <img class="carousel-food-spin__img" src="image/${image}" alt="dish ${image}">
+        </a>
+      </div>
```

Comment:
Probe algo y parece que no necesitas el `div`
```suggestion
      <a class="carousel-food-spin__main js-carousel-food-spin-dish" href="#" aria-label="click to see the selected dish">
        <img class="carousel-food-spin__img" src="image/${image}" alt="dish ${image}">
      </a>
```